bindings/python/setup.py

```diff
@@ -0,0 +1,143 @@
+import os
+import platform
+import subprocess
+import sys
+
+from setuptools import setup, Extension
+from setuptools.command.build_ext import build_ext
+from subprocess import check_call, check_output
+from distutils.spawn import find_executable
```

Comment:
Please don't put `distutils` back in. This is deprecated behavior.

In general it seems like this is a lot of extra work to use `setuptools` when `scikit-build-core` already exists. https://scikit-build-core.readthedocs.io/

Reply:
I expect this is here instead of `shutil.which` (and `scikit-build-core` is not used) due to the desire to support Python 2. I'd at least make this conditional on `sys.version_info`, though!

Reply:
Classic scikit-build does support Python 2, though :wink:

Reply:
I'd like to avoid a dependency on scikit-build if possible, but I don't mind removing distutils if there's an alternative way, I'm only using `find_executable`, so should be easy to find a good alternative.

Reply:
`shutil.which` replaces `distutils.spawn.find_executable`, Python 3 only.

Reply:
c.f. https://github.com/xrootd/xrootd/commit/80890de66f66208346c3f3e81aca8209fb60a852 from https://github.com/xrootd/xrootd/pull/1585

Reply:
Okay, but you have a dependency on setuptools already. In Python 3.12.0b1 it's not pre-installed anymore, so it's on equal footing with scikit-build. ;)

Reply:
Ok, nice, I will use something similar to what's in 80890de to only use distutils if necessary.

Reply:
Fixed.